halo2_proofs/examples/cost-model.rs

```diff
@@ -1,53 +1,9 @@
-use std::{
-    cmp, fmt, iter,
-    num::ParseIntError,
-    str::FromStr,
-    time::{Duration, Instant},
-};
-
-use ff::Field;
-use group::{Curve, Group};
 use gumdrop::Options;
-use halo2_proofs::arithmetic::best_multiexp;
-use halo2curves::pasta::pallas;
-
-struct Estimator {
-    /// Scalars for estimating multiexp performance.
-    multiexp_scalars: Vec<pallas::Scalar>,
-    /// Bases for estimating multiexp performance.
-    multiexp_bases: Vec<pallas::Affine>,
-}
-
-impl fmt::Debug for Estimator {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "Estimator")
-    }
-}
-
-impl Estimator {
-    fn random(k: usize) -> Self {
-        let max_size = 1 << (k + 1);
-        let mut rng = rand_core::OsRng;
-
-        Estimator {
-            multiexp_scalars: (0..max_size)
-                .map(|_| pallas::Scalar::random(&mut rng))
-                .collect(),
-            multiexp_bases: (0..max_size)
-                .map(|_| pallas::Point::random(&mut rng).to_affine())
-                .collect(),
-        }
-    }
-
-    fn multiexp(&self, size: usize) -> Duration {
-        let start = Instant::now();
-        best_multiexp(&self.multiexp_scalars[..size], &self.multiexp_bases[..size]);
-        Instant::now().duration_since(start)
-    }
-}
+use halo2_proofs::dev::cost_model::*;
 
+/// CLI Options to build a circuit specifiction to measure the cost model of.
 #[derive(Debug, Options)]
-struct CostOptions {
+struct CliCostOptions {
```

Comment:
I'm not even sure is worth having to maintain code for the CLI tool. No one uses it at all. I think is much easier to get rid of it completelly. SO that we need to maintain less code.

WDYT @han0110 @kilic @davidnevadoc ?

Reply:
The less code the better. Specially for unused stuff (that's my view). But we can defer the decision to the future.

Reply:
I agree with @CPerezz to remove the cli tool

Reply:
Removed the CLI tool.